### ISSUE TITLE
memory issue with cudaMemcpy

### ISSUE THREAD

**Issue by Stevo15025** [Stevo15025/tsp_gpu#3]
Running the current code nothing is being evaluated. Running the code with `cuda-memcheck` gives a bunch of errors such as 

```
========= CUDA-MEMCHECK
========= Program hit cudaErrorInvalidValue (error 11) due to "invalid argument" on CUDA API call to cudaMemcpy. 
=========     Saved host backtrace up to driver entry point at error
=========     Host Frame:/usr/lib/x86_64-linux-gnu/libcuda.so.1 [0x2eea03]
=========     Host Frame:./tsp_cuda [0x3eacf]
=========     Host Frame:./tsp_cuda [0x30f6]
=========     Host Frame:/lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main + 0xf5) [0x21f45]
=========     Host Frame:./tsp_cuda [0x2a99]
```

**Comment by Stevo15025** [comment]
This was resolved [here](http://stackoverflow.com/questions/40598634/cudamemcpy-invalid-argument-in-simple-vector-example/40598940#40598940)

**Comment by Stevo15025** [comment]
Fixed in commit  e42be48
